crates/node/src/lib.rs

```diff
@@ -427,10 +429,18 @@ impl Node {
                 application_id: context.application_id,
             });
         };
+        let args_str = str::from_utf8(&payload).map_err(|e| {
+            error!(%e, "Failed to convert payload to UTF-8.");
+            CallError::InternalError
+        })?;
+        let args_json: ProposalRequest = serde_json::from_str(args_str).map_err(|e| {
+            error!(%e, "Failed to parse proposal request.");
+            CallError::InternalError
+        })?;
```

Comment:
this should be from `outcome.proposals` which is the outcome of the execution, not `payload` (which is the argument to the function call)

`outcome.proposals` is a `Map<Id, Vec<u8>>`, the `proposal` payload should be deserialized `Vec<ProposalAction>`, and a `Proposal` can be constructed.

a similar path for `outcome.approvals`